### ISSUE TITLE
Not able to get vmstorage metrics

### ISSUE THREAD

**Issue by sudhir-maurya** [VictoriaMetrics/VictoriaMetrics#1213]
I am using the VictoriaMetircs cluster version with 2 instance of each services running in standalone environment. 
Not able to fetch the metrics for vmstorage when I try using the endpoint https://hostname:8482/metrics. I am using NAS for remote storage. sometime the metrics endpoint works sometime not.
In Prometheus target, it shows context deadline exceeded.

**Comment by valyala** [comment]
@sudhir-maurya , could you look into `vmstorage` logs? Probably they contain some information on why the [`/metrics` endpoint at vmstorage](https://victoriametrics.github.io/Cluster-VictoriaMetrics.html#monitoring) times out.

**Comment by sudhir-maurya** [comment]
@valyala I couldn't get as such info from logs, but when I tried the metrics endpoint by stopping vmselect, it was working fine. I am guessing if this issue is due to huge query request from vmselect causing slow in response. let me know what all details/metrics I can provide. pasting the snippet of  logs pattern below: 

```
2021-04-13T04:03:33.746Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 90421 rows across 78994 blocks in 15.386 seconds at 5876 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/80898_69471_20210413040252.923_20210413040312.533_16752A8632FB30A1"; sizeBytes: 459059
2021-04-13T04:03:34.971Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/342930_100521_20210413035144.092_20210413035327.759_16752A8632FB303F" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:03:42.203Z        info    VictoriaMetrics/lib/mergeset/table.go:896       merged 614 items across 1 blocks in 23.266 seconds at 26 items/sec to "/VmstoreData/prometheus1/indexdb/167334651C6896DB/182_1_16752A59A3E8C784"; sizeBytes: 2446
2021-04-13T04:03:48.380Z        info    VictoriaMetrics/lib/mergeset/table.go:896       merged 911 items across 4 blocks in 59.114 seconds at 15 items/sec to "/VmstoreData/prometheus1/indexdb/167334651C6896DB/824_1_16752A59A3E8C782"; sizeBytes: 7160
2021-04-13T04:03:48.422Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 45086 rows across 45003 blocks in 24.486 seconds at 1841 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/37099_37016_20210413040303.078_20210413040319.864_16752A8632FB30A2"; sizeBytes: 300284
2021-04-13T04:03:59.610Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 86631 rows across 76259 blocks in 25.863 seconds at 3349 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/73208_62836_20210413040311.899_20210413040327.665_16752A8632FB30A3"; sizeBytes: 387412
2021-04-13T04:04:04.148Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 58111 rows across 58060 blocks in 23.462 seconds at 2476 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/46805_46754_20210413040309.885_20210413040329.685_16752A8632FB30A4"; sizeBytes: 380175
2021-04-13T04:04:04.465Z        info    VictoriaMetrics/lib/mergeset/table.go:896       merged 894 items across 1 blocks in 22.262 seconds at 40 items/sec to "/VmstoreData/prometheus1/indexdb/167334651C6896DB/221_1_16752A59A3E8C785"; sizeBytes: 2429
2021-04-13T04:04:15.162Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 77122 rows across 76199 blocks in 20.977 seconds at 3676 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/59610_58683_20210413040321.654_20210413040345.809_16752A8632FB30A6"; sizeBytes: 471452
2021-04-13T04:04:23.930Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 99481 rows across 84771 blocks in 24.320 seconds at 4090 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/32824_32824_20210413040333.520_20210413040353.224_16752A8632FB30A7"; sizeBytes: 172032
2021-04-13T04:04:25.349Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 125211 rows across 124853 blocks in 25.343 seconds at 4940 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/62960_62853_20210413040325.446_20210413040352.849_16752A8632FB30A9"; sizeBytes: 460490
2021-04-13T04:04:26.308Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 18000 rows across 17881 blocks in 26.302 seconds at 684 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/11617_11607_20210413040322.923_20210413040351.553_16752A8632FB30A8"; sizeBytes: 97517
2021-04-13T04:04:26.509Z        info    VictoriaMetrics/lib/mergeset/table.go:896       merged 934 items across 1 blocks in 22.044 seconds at 42 items/sec to "/VmstoreData/prometheus1/indexdb/167334651C6896DB/282_1_16752A59A3E8C787"; sizeBytes: 2987
2021-04-13T04:04:27.906Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/61383_54958_20210413040037.848_20210413040146.001_16752A8632FB3091" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:04:32.048Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/25846_24031_20210413035933.708_20210413040031.655_16752A8632FB3087" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:04:37.006Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/27088_23422_20210413040048.078_20210413040119.650_16752A8632FB308B" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:04:37.645Z        info    VictoriaMetrics/lib/storage/partition.go:1257   merged 97395 rows across 96780 blocks in 27.278 seconds at 3570 rows/sec to "/VmstoreData/prometheus1/data/small/2021_04/84241_68927_20210413040346.574_20210413040407.390_16752A8632FB30AA"; sizeBytes: 592478
2021-04-13T04:04:42.055Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/61556_61326_20210413040031.574_20210413040048.403_16752A8632FB3089" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:04:44.213Z        warn    VictoriaMetrics/lib/fs/dir_remover.go:85        failed to remove directory "/VmstoreData/prometheus1/data/small/2021_04/55539_35346_20210413040039.885_20210413040121.680_16752A8632FB308D" due to NFS lock; retrying later in 1.600 seconds
2021-04-13T04:04:46.843Z        info    VictoriaMetrics/lib/mergeset/table.go:896       merged 704 items across 3 blocks in 58.508 seconds at 12 items/sec to "/VmstoreData/prometheus1/indexdb/167334651C6896DB/639_1_16752A59A3E8C786"; sizeBytes: 5994
```

**Comment by sudhir-maurya** [comment]
Able to get the metrics, issue seems to be with NFS storage type causing lot of complications like connection pool problem, improper file permission problem etc. Moving to EXT4 storage type resolves all the vmstorage isues.
closing this issue now.
